sql/binlog.h

```diff
@@ -949,6 +948,8 @@ class MYSQL_BIN_LOG: public TC_LOG
       @retval !=0    Error
   */
   int get_gtid_executed(Sid_map *sid_map, Gtid_set *gtid_set);
+private:
+  void publish_global_status_variables(void) const;
```

Comment:
I think this is better than the previous one. Just note that apart from position you are also publishing binlog_global_snapshot_file. Maybe publish_position_for_global_snapshot_status, but publish_position_for_global_status is good too.